src/mmw/js/src/data_catalog/models.js

```diff
@@ -29,6 +29,23 @@ var Catalog = Backbone.Model.extend({
         error: '',
     },
 
+    searchIfNeeded: function(query, fromDate, toDate, bounds) {
+        var self = this,
+            isSameSearch = query === this.get('query') &&
+                           fromDate === this.get('fromDate') &&
+                           toDate === this.get('toDate') &&
+                           utils.formatBounds(bounds) === this.get('bbox');
+
+        if (!isSameSearch) {
```

Comment:
Ideally we would cancel `this.searchPromise` before starting a new one. Not sure how that can be done. Another alternative may be to chain the new search onto the old search.

If I'm misreading the code (which I could: promises are hard to follow), please do correct me.